posts/2022-02-15-ga-release-post.adoc

```diff
@@ -0,0 +1,192 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19545[OpenIdConnectClient cookies not getting deleted after logout]
++
+Previously, OpenId related cookies were accumulating upon multiple logins which were being used and not being cleared after logout. Instead, we would expect all cookies to be cleared after logout or on next login, instead of accumulating. This issue was caused by incomplete logic for validating SSO cookies in some flows and has now been fixed.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19831[The output of ./wlp/bin/productInfo featureInfo missing new lines]
++
+Previously, the output of `./wlp/bin/productInfo featureInfo` was missing new line characters.
+For example:
++
+[source]
+----
+acmeCA-2.0adminCenter-1.0appClientSupport-1.0appSecurity-1.0appSecurity-2.0appSecurity-3.0appSecurityClient-1.0audit-1.0batch-1.0batchManagement-1.0beanValidation-1.1beanValidation-2.0bells-1.0cdi-1.2cdi-2.0cloudant-1.0concurrent-1.0constra ...
+----
++
+There are now line breaks between each of the feature names, as shown below:
++
+[source]
+----
+acmeCA-2.0
+adminCenter-1.0
+appClientSupport-1.0
+appSecurity-1.0
+appSecurity-2.0
+appSecurity-3.0
+appSecurityClient-1.0
+...
+----
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/18941[NullPointerException in JSP after upgrade]
++
+A bug was introduced in 21.0.0.9 which caused apps to crash with a NullPointerException. This issue has now been fixed, as it was caused by a `taglib` referencing a `tld` file that didn't exist.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19860[Updating MicroProfile versions on server.xml causes issues with install manager]
++
+Previously, updating to MicroProfile 5.0 in the `server.xml` and `pom.xml` would cause issues with the install manager. After resolving the features which are needed, the `RepositoryResolver` has to create lists of which features to install. There's a list of everything needed for each requested feature and one for each auto-feature. When creating the install lists, it traversed the dependencies but stopped traversing when it founds a dependency which was installed. If `distributedMap-1.0` is installed but `distributedMapInternal-1.0` is not, it stops traversing dependencies when it hits `distributedMap-1.0` and `distributedMapInternal-1.0` is not added to the list. This bug was fixed by tracing down the entire dependency tree, building the list and finally removing any features which are already installed.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/19826[MP Fault Tolerance annotations at the class level of a Rest Client interface are ignored]
++
+A bug was discovered which led to MP Fault Tolerance annotations to be ignored when at the class level of a Rest Client interface. However it would be picked up if annotated at method level. This issue was fixed by
```

Comment:
```suggestion
A bug was discovered which led to MP Fault Tolerance annotations to be ignored when annotated at the class level of a Rest Client interface. However it would be picked up if annotated at method level. This issue was fixed by updating the rest client builder to check for fault tolerance annotations on the interface as well as on the method when deciding whether to add the `@FaultTolerance` annotation.
```